riscv/insns/c_sspopchk_x5.h

```diff
@@ -0,0 +1,35 @@
+require_extension(EXT_ZICFISS);
+require_extension(EXT_ZCA);
+
+reg_t s = STATE.mstatus->read();
+reg_t prev_prv = get_field(s, MSTATUS_MPP);
+reg_t mSSE = get_field(STATE.menvcfg->read(), MENVCFG_SSE);
+
+if (prev_prv != PRV_M && !mSSE)
+  throw trap_illegal_instruction(insn.bits());
```

Comment:
1. Should use current privilege and not previous privilege.
2. The sspopchk when not enabled reverts to being a zimop - does not cause illegal instruction.
The code should be:
  ```
xSSE = 
  if   priv == Machine 
  then false;
  else if priv == Supervisor 
  then menvcfg.SSE() == 0b1;
  else if haveSupMode()
  then senvcfg.SSE() == 0b1;
  else false;
 if xSSE == FALSE
  << Zimop/Zcmop behavior>>
else
  <<Zicfiss defined behavior>>
```

Reply:
Thank you for your review. I think I can implement the following code.
```
if (prv == PRV_M)
  xSSE = 0;
else if (prv == PRV_S || prv == PRV_HS)
  xSSE = get_field(STATE.menvcfg->read(), MENVCFG_SSE);
else if (p->extension_enabled('S'))
  xSSE = get_field(STATE.senvcfg->read(), SENVCFG_SSE);
else
  xSSE = false;

// For VS-mode
if (prv == PRV_S && STATE.v)
  xSSE &= get_field(STATE.henvcfg->read(), HENVCFG_SSE);
```